aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt

```diff
@@ -37,28 +36,28 @@ sealed class AmplifyCredential {
 
     @Serializable
     @SerialName("userPool")
-    data class UserPool(override val signedInData: SignedInData) : AmplifyCredential(), UserPoolData
+    data class UserPool(override val signedInData: SignedInData) : AmplifyCredential(), UserPoolTypeCredential
 
     @Serializable
     @SerialName("identityPool")
     data class IdentityPool(override val identityId: String, override val credentials: AWSCredentials) :
-        AmplifyCredential(), IdentityPoolData
+        AmplifyCredential(), IdentityPoolTypeCredential
 
-//    @Serializable
-//    @SerialName("identityPoolFederated")
-//    data class IdentityPoolFederated(
-//        val federatedToken: FederatedToken,
-//        val identityId: String,
-//        val credentials: AWSCredentials
-//    ) : AmplifyCredential()
+    @Serializable
+    @SerialName("identityPoolFederated")
+    data class IdentityPoolFederated(
+        val federatedToken: FederatedToken,
+        val identityId: String?,
+        val credentials: AWSCredentials? = null
```

Comment:
identityId and credentials should be non-nullable.

Reply:
Good catch